src/mobile/components/WalletWithdrawBody/index.tsx

```diff
@@ -0,0 +1,142 @@
+        otpCode: '',
+        withdrawConfirmModal: false,
+        total: '',
+        withdrawDone: false,
+    });
+
+    const intl = useIntl();
+    const dispatch = useDispatch();
+    const user = useSelector(selectUserInfo);
+    const withdrawSuccess = useSelector(selectWithdrawSuccess);
+    const { currency, fee, type } = props.wallet;
+    const fixed = (props.wallet || { fixed: 0 }).fixed;
+    const withdrawAmountLabel = React.useMemo(() => intl.formatMessage({ id: 'page.body.wallets.tabs.withdraw.content.amount' }), [intl]);
+    const withdraw2faLabel = React.useMemo(() => intl.formatMessage({ id: 'page.body.wallets.tabs.withdraw.content.code2fa' }), [intl]);
+    const withdrawFeeLabel = React.useMemo(() => intl.formatMessage({ id: 'page.body.wallets.tabs.withdraw.content.fee' }), [intl]);
+    const withdrawTotalLabel = React.useMemo(() => intl.formatMessage({ id: 'page.body.wallets.tabs.withdraw.content.total' }), [intl]);
+    const withdrawButtonLabel = React.useMemo(() => intl.formatMessage({ id: 'page.body.wallets.tabs.withdraw.content.button' }), [intl]);
+
+    const isTwoFactorAuthRequired = (level: number, is2faEnabled: boolean) => {
+        return level > 1 || (level === 1 && is2faEnabled);
+    };
+    const getConfirmationAddress = () => {
+        let confirmationAddress = '';
+
+        if (props.wallet) {
+            confirmationAddress = props.wallet.type === 'fiat' ? (
+                withdrawData.beneficiary.name
+            ) : (
+                withdrawData.beneficiary.data ? (withdrawData.beneficiary.data.address as string) : ''
+            );
+        }
+
+        return confirmationAddress;
+    };
+    const toggleConfirmModal = (amount?: string, total?: string, beneficiary?: Beneficiary, otpCode?: string) => {
+        setWithdrawData((state: any) => ({
+            amount: amount || '',
+            beneficiary: beneficiary || defaultBeneficiary,
+            otpCode: otpCode || '',
+            withdrawConfirmModal: !state.withdrawConfirmModal,
+            total: total || '',
+            withdrawDone: false,
+        }));
+    };
+    const toggleSubmitModal = () => {
+        setWithdrawSubmitModal(state => !state);
+        setWithdrawData(state => ({ ...state, withdrawDone: true }));
+    };
+    const handleWithdraw = () => {
+        const { otpCode, amount, beneficiary } = withdrawData;
+        if (!props.wallet) {
+            return;
+        }
+
+        const withdrawRequest = {
+            amount,
+            currency: currency.toLowerCase(),
+            otp: otpCode,
+            beneficiary_id: String(beneficiary.id),
+        };
+        dispatch(walletsWithdrawCcyFetch(withdrawRequest));
+        toggleConfirmModal();
+    };
+
+    useWalletsAddressFetch(currency);
+    useBeneficiariesFetch();
+
+    const prevWithdrawSuccess = usePrevious(withdrawSuccess);
+
+    if (!prevWithdrawSuccess && withdrawSuccess) {
+        toggleSubmitModal();
+    }
```

Comment:
There is no need for additional hook here. What do you think?
```suggestion
React.useEffect(() => {
        if (withdrawSuccess) {
            toggleSubmitModal();
        }
}, [withdrawSuccess]);
```

Reply:
it is a useful hook to have. And in case **withdrawSuccess** changes from **true** to **false** it will call **toggleSubmitModal** and we don't want it